node/src/components/contract_runtime.rs

```diff
@@ -54,6 +58,19 @@ use metrics::Metrics;
 pub use operations::execute_finalized_block;
 pub(crate) use types::{BlockAndExecutionEffects, EraValidatorsRequest};
 
+use super::fetcher::FetchedOrNotFound;
+
+/// An enum that represents all possible error conditions of a `contract_runtime` component.
+#[derive(Debug, Error)]
+pub(crate) enum ContractRuntimeError {
+    /// The provided serialized id cannot be deserialized properly.
+    #[error("error deserializing id: {0}")]
+    InvalidSerializedId(Box<str>),
```

Comment:
This _could_ be just a bincode error, instead of being stringified, or maybe `Box<dyn Error + Send>` or similar. But I don't think it is a big issue if not.

In that case `#[source]` could also be set.

Reply:
Also, why `Box<str>` and not just `String`?

Reply:
`Box<str>` is smaller by 8 bytes.

Reply:
The changes to the error handling are here: https://github.com/casper-network/casper-node/pull/2614/commits/7bc53619fe7adaa9968a5e66fed2f8c168717586